gossip/emitter/ordering_test.go

```diff
@@ -194,3 +196,36 @@ func TestTransactionTimeSort(t *testing.T) {
 		}
 	}
 }
+
+func TestTransactionsOrdering_ZeroGasTip(t *testing.T) {
+	t.Parallel()
+
+	// This test ensures that transactions with zero gas tip do not overflow
+	// when calculating the miner fee for sorting purposes.
+
+	tx := types.NewTx(&types.DynamicFeeTx{
+		Nonce:     uint64(0),
+		To:        &common.Address{},
+		Value:     big.NewInt(100),
+		Gas:       100,
+		GasFeeCap: big.NewInt(50),
```

Comment:
could you make a similar test checking for transactions with zero GasPrice/GasFeeCap ?